### PR TITLE
internal/contributebot: keep PRs with "ready to submit" in sync with upstream

### PR DESCRIPTION
**Pull request by zombiezen**
This is an incremental (but significant) step toward accomplishing #687. It does not submit PRs, but it manages them so that they are more likely to be in a mergeable state. Since this is taking some shortcuts, I am not updating the design doc with details of this change yet, but this proves (to me at least) the feasibility of an approach where all state is stored in GitHub. Hopefully this eases enough maintainer toil that I can do design work for the remaining elements on a more relaxed schedule.

When a pull request is labeled with "ready to submit", Contribute Bot calls syncPullRequest to update the pull request. Using a label makes validation easy because only users with write access to a repository can make this change.

On any branch push to the repository, Contribute Bot will search for any open pull requests with the "ready to submit" label that have a base branch of the updated branch. Contribute Bot will call syncPullRequest for all the matching pull requests.

If any errors occur while merging, Contribute Bot will remove the "ready to submit" label and leave a comment. This gives notification of required manual intervention.

I've changed the built container to include a copy of Git. This does bump up the container to requiring debian-slim, but I don't see a good way around that.

Updates #687

### CODE REVIEW

File: internal/contributebot/process.go [hunk 1]
```diff
@@ -82,6 +96,16 @@ func hasLabel(iss *github.Issue, label string) bool {
 	return false
 }
 
+// prHasLabel returns true iff the issue has the given label.
```
Comment (vangent) [inline internal/contributebot/process.go]: Nit: issue -> pull request

**Comment by zombiezen** [comment]
Argh, this approach hit a snag. I was assuming that the "Maintainers Can Modify" would include GitHub applications installed on a repository, but a sample PR (https://github.com/zombiezen/go-cloud-bot-test/pull/9) shows that this is not the case. 🙁

This would mean that either Contribute Bot would need to obtain user credentials (act as a user) or we somehow find a set of GitHub APIs that are equivalent to pressing the "Update branch" button.

**Comment by zombiezen** [comment]
Alright, good news: I have a workaround for the auth issue. GitHub Applications can obtain [user-to-server credentials](https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/) so that the server can perform actions on behalf of a user instead of the application itself. I have manually tested that access tokens obtained in this way can be used to authorize pushes to branches in pull request forks that have the "Maintainers Can Modify" bit.

Bad news: things get more complicated. Instead of just keeping credentials for the application, Contribute Bot now needs to store credentials per maintainer and serve a 3-legged OAuth flow. When performing syncs, Contribute Bot needs to query for who requested the sync and authorize using their credentials.

**Comment by zombiezen** [comment]
Closing out this PR until the above changes are made.

**Pull request closed without merging**